src/litserve/server.py

```diff
@@ -100,16 +103,32 @@ def run_single_loop(lit_api, request_queue: Queue, request_buffer):
             pipe_s.send(y_enc)
 
 
-def inference_worker(
-    lit_api,
-    device,
-    worker_id,
-    request_queue,
-    request_buffer,
-    max_batch_size,
-    batch_timeout,
-):
+def run_streaming_loop(lit_api, request_queue: Queue, request_buffer):
+    while True:
+        try:
+            uid = request_queue.get(timeout=1.0)
```

Comment:
enabling batching would be straightforward here as you mentioned in earlier comment @lantiga. Would just have to collect the requests using `collate_request(...)` like [here](https://github.com/Lightning-AI/litserve/blob/a75e342fdc8a3248f72175b161c0c587f8187baf/src/litserve/server.py#L62). 

In this PR, the `streaming_loop` doesn't take any batching argument and hence won't add batching. I would like to enable that in a separate PR if you're okay with that.

Reply:
Ok let's keep it for later, but we need to produce an error message if one enables both batching and streaming. Also please add a corresponding test.

Reply:
Also please add open an issue for batching+streaming as part of addressing the review

Reply:
Created [here](https://github.com/Lightning-AI/litserve/issues/45)